### ISSUE TITLE
[Bug]: rspack转化ts结果不符合预期，typeof语句被转为0

### ISSUE THREAD

**Issue by sunft1996** [web-infra-dev/rspack#7001]
### System Info

 System:
    OS: macOS 13.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 36.62 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  
Binaries:
    Node: 16.20.2 - /usr/local/opt/nvm/versions/node/v16.20.2/bin/node
    Yarn: 1.22.19 - /usr/local/opt/nvm/versions/node/v16.20.2/bin/yarn
    npm: 8.19.4 - /usr/local/opt/nvm/versions/node/v16.20.2/bin/npm
    pnpm: 8.7.1 - /usr/local/opt/nvm/versions/node/v16.20.2/bin/pnpm
  
Browsers:
    Chrome: 126.0.6478.127
    Safari: 16.5
  
npmPackages:
    @rspack/cli: ^0.5.9 => 0.5.9 
    @rspack/core: ^0.5.9 => 0.5.9 

## rspack打包前

```JS
// src/index.ts
const xhr = {
    responseType: false,
    responseText: "OK",
    response: {}
}

console.log(
    ( xhr.responseType || "text" ) !== "text"  ||
    typeof xhr.responseText !== "string" ?
        { binary: 1 } :
        { text: 2 }
)

// 条件一 和 条件二 均为false，因此实际应该打印 { text: 2 }
```

## 打包后

```JS
// dist/main.js

var xhr = {
    responseType: false,
    responseText: "OK",
    response: {}
};
console.log((xhr.responseType || "text") !== "text" || 0 ? { // 语句：`typeof xhr.responseText !== "string"` 被转为 0 
    binary: 1
} : 0) // 此处打印{ binary: 1 } 或 0，结果都不对；

```

### Reproduce link

https://github.com/sunft1996/rspack-issue-ts-demo

### Reproduce Steps

```  shell
$ npm run bulid
```

**Comment by xc2** [comment]
Close as duplicated with #6124 that has been fixed by #6139

Please upgrade rspack to >= 0.6.0

**Comment by sunft1996** [comment]
> Close as duplicated with #6124 that has been fixed by #6139
> 
> Please upgrade rspack to >= 0.6.0

请问一下，为什么rspack内部会对js做转化？我并没有开启swc-loader，测试了一下webpack中默认也是没有对以上代码进行转化的。

**Comment by hardfist** [comment]
> > Close as duplicated with #6124 that has been fixed by #6139
> > Please upgrade rspack to >= 0.6.0
> 
> 请教一下，为什么rspack内部会对js做转化？我并没有开启swc-loader，测试了一下webpack中默认也是没有对以上代码进行转化的。

webpack其实内部会做一些代码转换，主要是基于 https://webpack.js.org/plugins/internal-plugins/#constplugin 的死代码删除优化， 只是rspack的对齐实现有bug，导致转换错误了
